From e4977baeb9fc07c91cb185b7a2f97695a560c0e6 Mon Sep 17 00:00:00 2001
From: Anders Berg <anders.berg@intel.com>
Date: Tue, 4 Mar 2014 11:40:47 +0100
Subject: [PATCH 016/278] net: ethernet: Add MDIO driver for LSI AXM55xx

Added driver for the MDIO controller found on LSI AXM55xx devices.

Signed-off-by: Anders Berg <anders.berg@intel.com>
---
 Documentation/devicetree/bindings/net/lsi-mdio.txt | 26 ++++++++++++++++++++++
 drivers/net/ethernet/Kconfig                       |  1 +
 drivers/net/ethernet/Makefile                      |  1 +
 3 files changed, 28 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/lsi-mdio.txt

diff --git a/Documentation/devicetree/bindings/net/lsi-mdio.txt b/Documentation/devicetree/bindings/net/lsi-mdio.txt
new file mode 100644
index 0000000..45b76359
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/lsi-mdio.txt
@@ -0,0 +1,26 @@
+* LSI Fast Ethernet Controller (FEMAC)
+
+Required properties:
+- compatible : Should be "lsi,axm-mdio"
+- reg : Address and length of the register set.
+- #address-cells : Must be 1.
+- #size-cells : Must be 0.
+
+Optional properties:
+- lsi,mdio-clk-offset : Set value used to offset the MDIO output signal. A
+value of N means the MDIO output signal is held for (N + 1) clk_per units long.
+- lsi,mdio-clk-period : Set value used to offset the MDC signal. A value of N
+means the MDC signal is 2 x (N + 1) clk_per units long.
+
+Example:
+
+mdio@2010090000 {
+	compatible = "lsi,axm-mdio";
+	reg = <0x20 0x10090000 0 0x1000>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	phy0: ethernet-phy@1e {
+		reg = <0x1e>;
+	};
+};
diff --git a/drivers/net/ethernet/Kconfig b/drivers/net/ethernet/Kconfig
index edae15ac..1f69f01 100644
--- a/drivers/net/ethernet/Kconfig
+++ b/drivers/net/ethernet/Kconfig
@@ -82,6 +82,7 @@ source "drivers/net/ethernet/ibm/Kconfig"
 source "drivers/net/ethernet/intel/Kconfig"
 source "drivers/net/ethernet/i825xx/Kconfig"
 source "drivers/net/ethernet/xscale/Kconfig"
+source "drivers/net/ethernet/lsi/Kconfig"
 
 config JME
 	tristate "JMicron(R) PCI-Express Gigabit Ethernet support"
diff --git a/drivers/net/ethernet/Makefile b/drivers/net/ethernet/Makefile
index bf7f450..2fa7916 100644
--- a/drivers/net/ethernet/Makefile
+++ b/drivers/net/ethernet/Makefile
@@ -48,6 +48,7 @@ obj-$(CONFIG_NET_VENDOR_XSCALE) += xscale/
 obj-$(CONFIG_JME) += jme.o
 obj-$(CONFIG_KORINA) += korina.o
 obj-$(CONFIG_LANTIQ_ETOP) += lantiq_etop.o
+obj-$(CONFIG_NET_VENDOR_LSI) += lsi/
 obj-$(CONFIG_NET_VENDOR_MARVELL) += marvell/
 obj-$(CONFIG_NET_VENDOR_MEDIATEK) += mediatek/
 obj-$(CONFIG_NET_VENDOR_MELLANOX) += mellanox/
-- 
1.8.5.2.233.g932f7e4

